timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/NamedMinTimestampLock.java

```diff
@@ -13,21 +13,26 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package com.palantir.atlasdb.timelock.lock;
 
 import com.palantir.lock.LockDescriptor;
-import com.palantir.lock.StringLockDescriptor;
 import java.util.UUID;
 import javax.ws.rs.NotSupportedException;
 
-public class ImmutableTimestampLock implements AsyncLock {
-
+final class NamedMinTimestampLock implements AsyncLock {
```

Comment:
These locks aren't really async, but the API needs that. The tests look weird and would probably throw off readers.

In some other PR, I can create a `SyncLock` interface and some wrapper that turns sync to async and use that here. Let me know if that sounds good?

Reply:
Don't think it's that important right now, but I agree it's weird